### DISCUSSION TITLE
BPFTRACE_MAX_PROBES increase and coredump occurred

### DISCUSSION THREAD

**Discussion by miloluo** [bpftrace/bpftrace#2104] [Q&A] [answered]
Hi,  expert
I wanna study some connection between the process comes from different different host.
Actually, I just wanna to break down the steps like this:
1. trace all the syscalls of those processes I'm interested .
2. find the key one or several syscalls I found from step 1.
3. trace and focus on those syscalls on step 2.

But now from the step 1 :
```
[root@dsc19c-n1 ~]# bpftrace -e 'tracepoint:syscalls:* { printf("%s [%d: %s]  \n", strftime("%H:%M:%S", nsecs), pid,comm); }'
ERROR: Can't attach to 612 probes because it exceeds the current limit of 512 probes.
You can increase the limit through the BPFTRACE_MAX_PROBES environment variable, but BE CAREFUL since a high number of probes attached can cause your system to crash.
[root@dsc19c-n1 ~]# export BPFTRACE_MAX_PROBES=620
[root@dsc19c-n1 ~]# bpftrace -e 'tracepoint:syscalls:* { printf("%s [%d: %s]  \n", strftime("%H:%M:%S", nsecs), pid,comm); }'
Attaching 612 probes...
bpftrace: /root/rpmbuild/BUILD/bpftrace/src/utils.cpp:122: void bpftrace::StdioSilencer::silence(): Assertion `old_stdio_ >= 0' failed.
Aborted (core dumped)
[root@dsc19c-n1 ~]# 
```

I get an error which says , there are 612 probes and currently I can only capture 512 unless I increase the limit with BPFTRACE_MAX_PROBES.

When I increase BPFTRACE_MAX_PROBES to 620, the program crash.

I am using RHEL 7.9 x86_64 and bpftrace is version of v0.13.0-124-g0971-dirty. 
```
[root@dsc19c-n1 ~]# cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 7.9 (Maipo)
[root@dsc19c-n1 ~]# 
[root@dsc19c-n1 ~]# bpftrace --version
bpftrace v0.13.0-124-g0971-dirty
[root@dsc19c-n1 ~]#

```

Could you help with this issue or is there any other way to workaround this?
Thanks very much.

**Comment by fbs** [comment]
might be due to an old bug we already fixed. Seems like you're using my centos 7 build. Ill try and get a new package out :)

**Reply by miloluo** [reply]
Hi, fbs
Yes， I am using the centos 7 build. 
Thanks very much.

**Reply by fbs** [reply]
can you try running `ulimit -n 4096` before executing the script? #2110

**Reply by miloluo** [reply]
Hi, fbs
Thanks for your reply.
With the ulimit value configured, the core dump issue is gone, but after several seconds, there are lots of processes lost events.

I issue another discussion about it: [Lost event occur when increasing the BPFTRACE_MAX_PROBES #2111](https://github.com/iovisor/bpftrace/discussions/2111)
Thanks very much.

**Comment by fbs** [comment, marked as answer]
Running into max open file issues, increasing the maximum helps. 

#2110 to fix it properly, for now `ulimit -n 4096; bpftrace ...`
